control-plane-operator/controllers/hostedcontrolplane/registryoperator/reconcile.go

```diff
@@ -193,6 +207,52 @@ func ReconcileDeployment(deployment *appsv1.Deployment, params Params) error {
 			},
 		)
 	}
+	// For ARO HCP deployments, we pass environment variables so we authenticate with Azure API through certificate
+	// authentication. We also mount the SecretProviderClass for the Secrets Store CSI driver to use; it will grab the
+	// certificate related to the ARO_HCP_MI_CLIENT_ID and mount it as a volume in the ingress pod in the path,
+	// ARO_HCP_CLIENT_CERTIFICATE_PATH.
+	if azureutil.IsAroHCP() {
+		deployment.Spec.Template.Spec.Containers[0].Env = append(deployment.Spec.Template.Spec.Containers[0].Env,
+			corev1.EnvVar{
+				Name:  "ARO_HCP_MI_CLIENT_ID",
```

Comment:
Also I'd drop ARO_ from the naming since cert auth is building block we might choose to leverage beyond aro. Thought I understand this might be problematic now, so feel free to ignore

Reply:
Will do on the consts in a single place.

Yeah we'd need to update all the OpenShift component PRs again if we wanted to drop the ARO part so it is problematic to change that now.